modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1525,7 +1534,89 @@ private Predicate _fillAccountEntriesPredicate(
 		Column<?, Long> column = (Column<?, Long>)table.getColumn(
 			objectField.getDBColumnName());
 
-		return column.in(ArrayUtil.toLongArray(accountEntryIds));
+		JoinStep joinStep = DSLQueryFactoryUtil.selectDistinct(
```

Comment:
I think we shouldn't use select Distinct in both cases, as it can affect the performance. For the first if condition, it just lists all entries of the table filtered by companyId and status, so we are not going to have duplicates.
Please let me know if I'm wrong.

Reply:
You are right.. I changed it to `select`